test/e2e/pull_test.go

```diff
@@ -172,7 +172,7 @@ var _ = Describe("Podman pull", func() {
 		session := podmanTest.Podman([]string{"pull", "-q", "--authfile", "/tmp/nonexistent", ALPINE})
 		session.WaitWithDefaultTimeout()
 		Expect(session).To(ExitWithError())
-		Expect(session.ErrorToString()).To(Equal("Error: credential file is not accessible: stat /tmp/nonexistent: no such file or directory"))
+		Expect(session.ErrorToString()).To(MatchRegexp("Error: credential file is not accessible: (faccessat|stat) /tmp/nonexistent: no such file or directory"))
```

Comment:
Why can this both? This makes testing the error message much more complicated.

Reply:
wanted to reduce tests failures in case we end up bisecting c/storage, but probably not worth the cost of dealing with regex.

Reply:
We do not run these outside of linux so not really IMO. I really do not like to add additional complexity to the tests.
For example see https://github.com/containers/podman/pull/22346 where it would add quite a bit of work to add regex logic.

Reply:
I still think this is a problem, I don't like having to write regex for all the error messages even when we know we always expect faccessat as we only run the tests on linux but I like to know what @edsantiago thinks about this.

Reply:
Under what circumstances will tests see `stat`? I too am *strongly* against the use of regex. It will disasterify a lot of my upcoming test cleanup.

Reply:
Never, my understanding stat is only used on native windows platforms.  Everwhere else it uses faccess (Linux, Darwin) Probably freebsd as well.